File: algorithm/python/sort/bubble_sort.py
import random

def bubbleSort(arr):
    for i in range(0, len(arr)-1):
        isSwap = False
        for j in range(0, len(arr) - 1 - i):
            if arr[j] > arr[j+1]: 
                arr[j+1], arr[j] = arr[j], arr[j+1]
                isSwap = True
        if isSwap == False: break 
    return arr

arr = random.sample(range(100), 20)
print(bubbleSort(arr))

File: algorithm/javascript/sort/bubble_sort.js
function bubbleSort(list) {
  for (let i = 0; i < list.length - 1; i++) {
    let flag = false;
    for (let j = 0; j < list.length - 1 - i; j++) {
      if (list[j] > list[j + 1]) {
        [list[j], list[j + 1]] = [list[j + 1], list[j]];
        flag = true;
      }
    }
    if (!flag) break;
  }
  return list;
}

const list = new Array(10).fill(1);
list.some((e, i) => {
  list[i] = e * Math.random() * 19 + 1;
});
console.log(bubbleSort(list));
